Struggling a bit with this. I tried something like
```diff
diff --git a/torchgeo/datasets/skyscript.py b/torchgeo/datasets/skyscript.py
index 916e070c6..a6772a315 100644
--- a/torchgeo/datasets/skyscript.py
+++ b/torchgeo/datasets/skyscript.py
@@ -79,6 +79,7 @@ class SkyScript(NonGeoDataset):
         transforms: Callable[[Sample], Sample] | None = None,
         download: bool = False,
         checksum: bool = False,
+        tokenizer: 'tokenizers.models.Model | None' = None,
     ) -> None:
         """Initialize a new SkyScript instance.
 
@@ -89,6 +90,7 @@ class SkyScript(NonGeoDataset):
                 entry and returns a transformed version.
             download: If True, download dataset and store it in the root directory.
             checksum: If True, check the MD5 of the downloaded files (may be slow).
+            tokenizer: A pre-trained tokenizer (defaults to BPE).
 
         Raises:
             AssertionError: If *split* is invalid.
@@ -96,7 +98,6 @@ class SkyScript(NonGeoDataset):
             DependencyNotFoundError: If tokenizers is not installed.
         """
         assert split in self.caption_files
-        tokenizers = lazy_import('tokenizers')
 
         self.root = pathlib.Path(root)
         self.split = split
@@ -107,11 +108,15 @@ class SkyScript(NonGeoDataset):
         self._verify()
 
         self.captions = pd.read_csv(self.root / self.caption_files[split])
-        train_captions = pd.read_csv(self.root / self.caption_files['train'])
 
-        self.tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE())
-        trainer = tokenizers.trainers.BpeTrainer()
-        self.tokenizer.train_from_iterator(train_captions['title'], trainer)
+        if tokenizer is None:
+            tokenizers = lazy_import('tokenizers')
+            self.tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE())
+            trainer = tokenizers.trainers.BpeTrainer()
+            train_captions = pd.read_csv(self.root / self.caption_files['train'])
+            self.tokenizer.train_from_iterator(train_captions['title'], trainer)
+        else:
+            self.tokenizer = tokenizer
 
     def __len__(self) -> int:
         """Return the number of images in the dataset.
```
The problem is that there's no way to add proper type hints for `tokenizer` because tokenizers isn't a required dependency. Best workaround I can think of is to ignore ruff/ty complaints, but that doesn't sound like a good idea. I would rather stick to hard-coding BPE if necessary. Any other options I'm missing?

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -177,6 +177,7 @@ intersphinx_mapping = {
     'shapely': ('https://shapely.readthedocs.io/en/stable/', None),
     'sklearn': ('https://scikit-learn.org/stable/', None),
     'timm': ('https://huggingface.co/docs/timm/main/en/', None),
+    'tokenizers': ('https://huggingface.co/docs/tokenizers/main/en/', None),
     'torch': ('https://docs.pytorch.org/docs/stable/', None),
     'torchmetrics': ('https://lightning.ai/docs/torchmetrics/stable/', None),
     'torchvision': ('https://docs.pytorch.org/vision/stable/', None),

--- a/tests/datasets/test_skyscript.py
+++ b/tests/datasets/test_skyscript.py
@@ -13,7 +13,7 @@ from torch import Tensor
 
 from torchgeo.datasets import DatasetNotFoundError, SkyScript
 
-pytest.importorskip('tokenizers', minversion='0.14')
+tokenizers = pytest.importorskip('tokenizers', minversion='0.14')
 
 
 class TestSkyScript:
@@ -32,6 +32,14 @@ class TestSkyScript:
 
     def test_len(self, dataset: SkyScript) -> None:
         assert len(dataset) == 2
+
+    def test_tokenizer(self, dataset: SkyScript) -> None:
+        tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE())
+        dataset = SkyScript(dataset.root, tokenizer=tokenizer)
+        x = dataset[0]
+        assert isinstance(x, dict)
+        assert isinstance(x['image'], Tensor)
+        assert isinstance(x['caption'], Tensor)
 
     def test_already_downloaded(self, dataset: SkyScript) -> None:
         shutil.rmtree(os.path.join(dataset.root, 'images2'))

--- a/torchgeo/datasets/skyscript.py
+++ b/torchgeo/datasets/skyscript.py
@@ -6,7 +6,7 @@
 import pathlib
 import textwrap
 from collections.abc import Callable
-from typing import ClassVar, Literal
+from typing import TYPE_CHECKING, ClassVar, Literal
 
 import numpy as np
 import pandas as pd
@@ -26,6 +26,9 @@ from .utils import (
     extract_archive,
     lazy_import,
 )
+
+if TYPE_CHECKING:
+    import tokenizers
 
 
 class SkyScript(NonGeoDataset):
@@ -79,6 +82,7 @@ class SkyScript(NonGeoDataset):
         transforms: Callable[[Sample], Sample] | None = None,
         download: bool = False,
         checksum: bool = False,
+        tokenizer: 'tokenizers.models.Model | None' = None,
     ) -> None:
         """Initialize a new SkyScript instance.
 
@@ -89,14 +93,18 @@ class SkyScript(NonGeoDataset):
                 entry and returns a transformed version.
             download: If True, download dataset and store it in the root directory.
             checksum: If True, check the MD5 of the downloaded files (may be slow).
+            tokenizer: A pre-trained tokenizer
+                (defaults to :class:`~tokenizers.models.BPE`).
 
         Raises:
             AssertionError: If *split* is invalid.
             DatasetNotFoundError: If dataset is not found and *download* is False.
             DependencyNotFoundError: If tokenizers is not installed.
+
+        .. versionadded:: 0.10
+           The *tokenizer* parameter.
         """
         assert split in self.caption_files
-        tokenizers = lazy_import('tokenizers')
 
         self.root = pathlib.Path(root)
         self.split = split
@@ -107,11 +115,15 @@ class SkyScript(NonGeoDataset):
         self._verify()
 
         self.captions = pd.read_csv(self.root / self.caption_files[split])
-        train_captions = pd.read_csv(self.root / self.caption_files['train'])
 
-        self.tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE())
-        trainer = tokenizers.trainers.BpeTrainer()
-        self.tokenizer.train_from_iterator(train_captions['title'], trainer)
+        if tokenizer is None:
+            tokenizers = lazy_import('tokenizers')
+            self.tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE())
+            trainer = tokenizers.trainers.BpeTrainer()
+            train_captions = pd.read_csv(self.root / self.caption_files['train'])
+            self.tokenizer.train_from_iterator(train_captions['title'], trainer)
+        else:
+            self.tokenizer = tokenizer
 
     def __len__(self) -> int:
         """Return the number of images in the dataset.

